### PR TITLE
feat: show warning dialog for unsupported versions

### LINKED ISSUE #866: Add dialog if you are using an unrecommended version.
### Type

Functionality

### Issue

Some noobs don't understand that a specific version is required for patches to apply properly. 

### Feature

When you select an apk/app that is not the suggested version there would be a popup saying like:

"The version you selected is not suggested to use. Patching errors may occur. Please use the suggested version."

It would be even better if it said what the suggested version is for that application they are attempting to patch.

Additionally, you can make it so that if they press "OK" it will disable Version compatibility check.

### Motivation

It will help the less advanced users and lower the amount of posts and comments that they ask for assistance. 

### Additional context

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] The issue is related solely to the ReVanced Manager

### PR DESCRIPTION
**Pull request by andrisas**
closes #866

### CODE REVIEW

File: assets/i18n/en_US.json [hunk 1]
```diff
@@ -49,7 +49,8 @@
         "widgetTitle": "Patcher",
         "patchButton": "Patch",
         "patchDialogText": "You have selected a resource patch and a split APK installation has been detected, so patching errors may occur.\nAre you sure you want to proceed?",
-        "armv7WarningDialogText": "Patching on ARMv7 devices is not yet supported and might fail. Proceed anyways?"
+        "armv7WarningDialogText": "Patching on ARMv7 devices is not yet supported and might fail. Proceed anyways?",
+        "versionWarningDialogText": "Patching this version is not recomended and might fail. Proceed anyways?"
```
Comment (Ushie) [inline assets/i18n/en_US.json:53]:
```diff
- recomended
+ recommended
```

**Comment by Ushie** [comment]
If I'm an armv7 user, this dialog will never be shown to me

**Review by Ushie — CHANGES_REQUESTED** [review]
If I'm an armv7 user, this dialog will never be shown to me

**Review by Ushie — CHANGES_REQUESTED** [review]
Universal and experimental patches shouldn't work as a bypass to this dialog

**Review by Ushie — CHANGES_REQUESTED** [review]
Can you do a few things?

- [ ] show this when selecting the application, instead of starting patching
- [ ] include the supported version in the dialog somewhere
- [ ] maybe show select from storage instead of continue anyways? I'm not sure about that so let's wait for @Aunali321 on that

don't revert your changes refactoring the armv7 dialog to showWarningDialogs, its likely to be of use in the future

**Comment by Aunali321** [comment]
If you go with restricting, you should account for the entire compatibility list, as the latest suggested version isnt the only supported one

**Review by Ushie — CHANGES_REQUESTED** [review]
Over the course of this commit, "recommended" is mistyped as "recomended"

File: assets/i18n/en_US.json [hunk 2]
```diff
@@ -50,7 +50,7 @@
         "patchButton": "Patch",
         "patchDialogText": "You have selected a resource patch and a split APK installation has been detected, so patching errors may occur.\nAre you sure you want to proceed?",
         "armv7WarningDialogText": "Patching on ARMv7 devices is not yet supported and might fail. Proceed anyways?",
-        "versionWarningDialogText": "Patching this version is not recommended and might fail. Proceed anyways?"
+        "versionWarningDialogText": "Patching this version is not recommended and might fail. The recomended version is {recomendedVer}. Proceed anyways?"
```
Comment (Ushie) [inline assets/i18n/en_US.json:53]: Over the course of this commit, "recommended" is miswrote as "recomended"

**Review by Ushie — CHANGES_REQUESTED** [review]

File: lib/ui/views/app_selector/app_selector_viewmodel.dart [hunk 3]
```diff
@@ -97,11 +98,51 @@ class AppSelectorViewModel extends BaseViewModel {
...
         Navigator.pop(context);
       }
     }
   }
 
+  Future showVersionNotSupportedDialog(
+    BuildContext context,
+  ) async {
+    final recommendedVersion = await _patcherAPI.getSuggestedVersion(
+        locator<PatcherViewModel>().selectedApp!.packageName);
```
Comment (Ushie) [inline lib/ui/views/app_selector/app_selector_viewmodel.dart:112]: Could you check whether the installed version matches any of the versions in the compatibility list?

**Pull request closed without merging**